site/docs/plugins/chat-members.md

```diff
@@ -0,0 +1,132 @@
+instance of any class that implements the [`StorageAdapter`](https://deno.land/x/grammy/mod.ts?s=StorageAdapter) interface.
+
+<CodeGroup>
+  <CodeGroupItem title="TypeScript" active>
+
+```ts
+import { Bot, Context, MemorySessionStorage } from "grammy";
+import type { ChatMember } from "grammy/types";
+import { chatMembers, ChatMembersFlavor } from "@grammyjs/chat-members";
+
+type MyContext = Context & ChatMembersFlavor;
+
+const adapter = new MemorySessionStorage<ChatMember>();
+
+const bot = new Bot<MyContext>("<your bot token>");
+
+bot.use(chatMembers(adapter));
+
+bot.start({
+  allowed_updates: ["chat_member", "message"],
+  onStart: ({ username }) => console.log(`Listening as ${username}`),
+});
+```
+
+</CodeGroupItem>
+
+<CodeGroupItem title="JavaScript" active>
+
+```js
+import { Bot, Context, MemorySessionStorage } from "grammy";
+import { chatMembers, ChatMembersFlavor } from "@grammyjs/chat-members";
+
+const adapter = new MemorySessionStorage();
+
+const bot = new Bot("<your bot token>");
+
+bot.use(chatMembers(adapter));
+
+bot.start({
+  allowed_updates: ["chat_member", "message"],
+  onStart: ({ username }) => console.log(`Listening as ${username}`),
+});
+```
+
+</CodeGroupItem>
+
+<CodeGroupItem title="Deno" active>
+
+```ts
+import {
+  Bot,
+  Context,
+  MemorySessionStorage,
+} from "https://deno.land/x/grammy/mod.ts";
+import type { ChatMember } from "https://deno.land/x/grammy/types.ts";
+import {
+  chatMembers,
+  ChatMembersFlavor,
+} from "https://deno.land/x/grammy_chat_members/mod.ts";
+
+type MyContext = Context & ChatMembersFlavor;
+
+const adapter = new MemorySessionStorage<ChatMember>();
+
+const bot = new Bot<MyContext>("<your bot token>");
+
+bot.use(chatMembers(adapter));
+
+bot.start({
+  allowed_updates: ["chat_member", "message"],
+  onStart: ({ username }) => console.log(`Listening as ${username}`),
+});
+```
+
+</CodeGroupItem>
+
+</CodeGroup>
+
+### Reading Chat Members
+
+This plugin also adds a new `ctx.chatMembers.getChatMember` function that will check the storage for information about a chat member before querying Telegram for it.
+If the chat member exists in the storage, it will be returned.
+Otherwise, `ctx.api.getChatMember` will be called and the result will be saved to the storage, making subsequent calls faster and removing the need to call Telegram again for that user and chat in the future.
+
+Here's an example:
+
+```ts
+bot.on("message", async (ctx) => {
+  const chatMember = await ctx.chatMembers.getChatMember();
+
+  return ctx.reply(
+    `Hello, ${chatMember.user.first_name}! I see you are a ${chatMember.status} of this chat!`,
+  );
+});
+```
+
+All parameters are optional; if you don't provide them, `ctx.chat.id` and `ctx.from.id` will be used instead.
+Please notice that, if you don't provide a chat id and there's no `chat` property inside the context (for example, on inline query updates), this will throw an error.
```

Comment:
```suggestion
Please notice that, if you don't provide a chat id and there's no `ctx.chat` property (for example, on inline query updates), this will throw an error.
```